frame/fast-unstake/src/weights.rs

```diff
@@ -62,37 +63,41 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:1 w:0)
 	// Storage: FastUnstake Head (r:1 w:1)
 	// Storage: Staking CurrentEra (r:1 w:0)
-	// Storage: Staking SlashingSpans (r:1 w:0)
-	// Storage: Staking Bonded (r:1 w:1)
-	// Storage: Staking Validators (r:1 w:0)
-	// Storage: Staking Nominators (r:1 w:0)
-	// Storage: System Account (r:1 w:1)
-	// Storage: Balances Locks (r:1 w:1)
-	// Storage: Staking Ledger (r:0 w:1)
-	// Storage: Staking Payee (r:0 w:1)
+	// Storage: Staking SlashingSpans (r:128 w:0)
+	// Storage: Staking Bonded (r:128 w:128)
+	// Storage: Staking Validators (r:128 w:0)
+	// Storage: Staking Nominators (r:128 w:0)
+	// Storage: System Account (r:128 w:128)
+	// Storage: Balances Locks (r:128 w:128)
+	// Storage: Staking Ledger (r:0 w:128)
+	// Storage: Staking Payee (r:0 w:128)
 	fn on_idle_unstake() -> Weight {
-		// Minimum execution time: 82_426 nanoseconds.
-		Weight::from_ref_time(83_422_000 as u64)
-			.saturating_add(T::DbWeight::get().reads(11 as u64))
-			.saturating_add(T::DbWeight::get().writes(6 as u64))
+		// Minimum execution time: 4_975_250 nanoseconds.
+		Weight::from_ref_time(5_072_221_000)
+			.saturating_add(T::DbWeight::get().reads(773))
+			.saturating_add(T::DbWeight::get().writes(641))
```

Comment:
Lot more Reads and Writes.

Reply:
is actually correct, the previous weight must have been out of date. You can see in `Polkadot` it is correct (the value is correlated with `BatchSize`):

https://github.com/paritytech/polkadot/blob/f4b8c910838ab65a0ba909c70e611dda16d4a31d/runtime/polkadot/src/weights/pallet_fast_unstake.rs#L54

Reply:
Too pessimistic, I will make the benchmark be a function of the actual batch size.